### PR TITLE
Support hidden and current venv dirs, pyenv locals, add tests

### PR DESCRIPTION
**Pull request by scop**

### CODE REVIEW

**Review by guludo — CHANGES_REQUESTED** [review]
Hi there.

Thank you for this! I like the changes.

The test is passing with python 3.6 but it is failing for python 3.8 (the versions I have in my current environment). Here is the error message:

```
$ python -m unittest
E
======================================================================
ERROR: testGuess (tests.test_venvrun.VenvRunTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/guludo/foss/venv-run/tests/test_venvrun.py", line 28, in testGuess
    with patch("venvrun.platform.system", return_value='Linux'), \
  File "/usr/lib/python3.8/unittest/mock.py", line 1377, in __enter__
    self.target = self.getter()
  File "/usr/lib/python3.8/unittest/mock.py", line 1552, in <lambda>
    getter = lambda: _importer(target)
  File "/usr/lib/python3.8/unittest/mock.py", line 1228, in _importer
    thing = _dot_lookup(thing, comp, import_path)
  File "/usr/lib/python3.8/unittest/mock.py", line 1217, in _dot_lookup
    __import__(import_path)
ModuleNotFoundError: No module named 'venvrun.subprocess'; 'venvrun' is not a package

----------------------------------------------------------------------
Ran 1 test in 0.001s

FAILED (errors=1)
```

Also, about the commit adding support when the current working directory is the environment, I'm not so sure about merging that one. What is the use-case for that?

Best regards,
Gustavo Sousa

File: MANIFEST.in [hunk 1]
```diff
@@ -0,0 +1 @@
+recursive-include tests *.py *python
```
[the remark this answers is not in the document]
Comment (scop) [reply MANIFEST.in:1]: I don't think that's something we want to be including in the dist (nor git).

**Comment by scop** [comment]
> Also, about the commit adding support when the current working directory is the environment, I'm not so sure about merging that one. What is the use-case for that?

The same as for all others: some people like to set up their venvs to that dir, e.g. `python -m venv .`. Incidentally, my main motivation for these changes is to replace a "homebrew" implementation [I did for Home Assistant](https://github.com/home-assistant/core/blob/e601f6241625a7051abad456fa978efb14e8c986/script/run-in-env.sh) with a more generic one and to be able to use it with other similarly affected projects. And the founder of Home Assistant [likes his venv in the project root](https://github.com/home-assistant/core/pull/30922#issuecomment-581038941), meaning if that part is dropped here, I'll be missing my goal at least partially ;)

File: MANIFEST.in (hunk 1 above)
[the remark this answers is not in the document]
Comment (guludo) [reply MANIFEST.in:1]: Well, the test code expects it to be present, right? (https://github.com/guludo/venv-run/pull/4/commits/764982985eef9ca4a710baeec73040dfce0777db#diff-19c5fcead6667a15d660dd0fae71fe76c87b4ab486d1a4b50dc35ef46fde8060L20-R32)

**Comment by scop** [comment]
> The test is passing with python 3.6 but it is failing for python 3.8 (the versions I have in my current environment).

Curious. It passes for me with all versions I've tried: 3.5.7, 3.6.9, 3.8.5, and 3.9.1. I'll see if I can figure out something.

**Comment by guludo** [comment]
> > Also, about the commit adding support when the current working directory is the environment, I'm not so sure about merging that one. What is the use-case for that?
> 
> The same as for all others: some people like to set up their venvs to that dir, e.g. `python -m venv .`. Incidentally, my main motivation for these changes is to replace a "homebrew" implementation [I did for Home Assistant](https://github.com/home-assistant/core/blob/e601f6241625a7051abad456fa978efb14e8c986/script/run-in-env.sh) with a more generic one and to be able to use it with other similarly affected projects. And the founder of Home Assistant [likes his venv in the project root](https://github.com/home-assistant/core/pull/30922#issuecomment-581038941), meaning if that part is dropped here, I'll be missing my goal at least partially ;)

Oh, I see. Okay, then :-)

File: MANIFEST.in (hunk 1 above)
[the remark this answers is not in the document]
Comment (scop) [reply MANIFEST.in:1]: Oops! You're absolutely right, it's _that_ `.python-version`! Will fix.

**Comment by guludo** [comment]
> > The test is passing with python 3.6 but it is failing for python 3.8 (the versions I have in my current environment).
> 
> Curious. It passes for me with all versions I've tried: 3.5.7, 3.6.9, 3.8.5, and 3.9.1. I'll see if I can figure out something.

Hm.. I wonder if that is a problem with my version then... I'll check here.

**Comment by guludo** [comment]
> Hm.. I wonder if that is a problem with my version then... I'll check here.


Okay. The error with the was caused by my stupidity haha

I already had a version of venvrun installed (in `.local/lib/python3.8/site-packages/`) and that one was being loaded by the test script. I should have started a new virtual environment before running the test. With a new environment the test works fine. :-)

**Review by guludo — APPROVED** [review]

**Comment by guludo** [comment]
I'll apply this now.

**Comment by scop** [comment]
Seems to work great for my use case, eagerly waiting for a PyPI release :)

**Comment by guludo** [comment]
Done! https://pypi.org/project/venv-run/0.1.0/ :-)

**Pull request merged**